packages/aws-cdk-lib/core/test/stack.test.ts

```diff
@@ -528,6 +529,64 @@ describe('stack', () => {
     });
   });
 
+  test('cross-stack references of nested stack lists returned from Fn::GetAtt work', () => {
+    // GIVEN
+    const app = new App();
+    const producer = new Stack(app, 'Producer');
+    const nested = new NestedStack(producer, 'Nestor');
+    const exportResource = new CfnResource(nested, 'exportedResource', {
+      type: 'BLA',
+    });
+    const consumer = new Stack(app, 'Consumer');
+    // L1s represent attribute names with `attr${attributeName}`
+    (exportResource as any).attrList = ['magic-attr-value'];
+
+    // WHEN - used in another stack
+    new CfnResource(consumer, 'SomeResource', {
+      type: 'BLA',
+      properties: {
+        Prop: exportResource.getAtt('List', ResolutionTypeHint.STRING_LIST),
+      },
+    });
```

Comment:
Probably a good idea to make sure number and string outputs still work as expected with nested stacks

Reply:
The test `automatic cross-stack references and manual exports look the same: nested stack edition` already covers this behavior well.  Number versus string outputs work exactly the same and number parameters are a very special case.

Functionally I went ahead and added both to the integ test I added since we didn't have it before and that'll be the most effective way to test them for certain.